src/Features/LanguageServer/Protocol/Handler/IDocumentChangeTracker.cs

```diff
@@ -16,14 +18,14 @@ namespace Microsoft.CodeAnalysis.LanguageServer.Handler;
 /// </summary>
 internal interface IDocumentChangeTracker
 {
-    ValueTask StartTrackingAsync(Uri documentUri, SourceText initialText, CancellationToken cancellationToken);
+    ValueTask StartTrackingAsync(Uri documentUri, SourceText initialText, LanguageInformation? languageInformation, CancellationToken cancellationToken);
```

Comment:
If the user manually changes the type, how does that get represented? A close + reopen with a new language?

Reply:
yup!